store/types/validity.go

```diff
@@ -1,15 +1,30 @@
 package types
 
+import math "math"
+
+const (
+	// 32K
+	MaxKeyLength = math.MaxUint16
```

Comment:
`MaxUint16` is picked so it can be encoded in 2bytes, should be more than enough for all use cases.

Reply:
This claim must be substantiated with empirical data, otherwise it can break consumers

Reply:
2^16 bytes (64 KiB) is definitely not enough for CosmWasm values

Reply:
Are there any length limits for keys on DBs used that are shorter than 2^32? Personally, I don't see any reasons why we should have a shorter limit for keys than for values. The original motivation was deterministic behaviour.

Reply:
👆 Data structures like sets (i.e. a map from key to nothing) store all its data in keys. So treating keys and value differently should be avoided if possible IMO.

Reply:
> 👆 Data structures like sets (i.e. a map from key to nothing) store all its data in keys. So treating keys and value differently should be avoided if possible IMO.

A set in this sense is an outlier use case and not something that the core KVStore interface should accommodate. A key-value store will in general restrict the key size to be far smaller than the value size.